File: Soal1.java
/*
 Nama : Wilson Angelie Tan, Stan F., Theophilus Samuel G.
 NPM :140810230024, 140810230046, 140810230054
 Tanggal Buat :13/10/2024
 Deskripsi :Cari nilai rata-rata nilai, nilai
           terbesar dan terkecilmenggunakan array satu dimensi dalam
           bentuk OOP 2 class bahasapython
*/

import java.util.Scanner;

class Logika {
    static Scanner sc = new Scanner(System.in);

    private int[] arr;
    private int ukuran;

    public Logika(){
        this.ukuran = 10;
        this.arr = new int[ukuran];
    }

    public Logika(int ukuran){
        this.arr = new int[ukuran];
    }

    public void setUkuran(int ukuran) {
        this.ukuran = ukuran;
        this.arr = new int[this.ukuran];
    }

    public void inputUkuran(){
        System.out.print("Masukkan Ukuran Array : ");
        this.ukuran = sc.nextInt();
        this.arr = new int[this.ukuran];
    }

    public int getUkuran() {
        return ukuran;
    }

    public void setArr(int[] arr) {
        this.arr = arr;
    }

    public int[] getArr() {
        return arr;
    }

    public void isiArr(){
        for(int a = 0; a < this.ukuran; a++){
            System.out.print("Masukkan nilai ke-" + (a+1) + " = ");
            arr[a] = sc.nextInt();
        }
    }

    public float cariRata2(){
        float jumlah = 0;

        for(int a : arr){ // foreach
            jumlah += a;
        }

        return jumlah/this.ukuran;
    }

    public int nilaiMax(){
        int highest = -99999;
        for(int a : arr){
            if(highest < a)
                highest = a;
        }
        return highest;
    }

    public int nilaiMin(){
        int lowest = 99999;
        for(int a : arr){
            if(lowest > a)
                lowest = a;
        }
        return lowest;
    }

    public void jumlahMatriks(Logika m1, Logika m2){
        if(m1.ukuran != m2.ukuran){
            System.out.println("Ukuran Larik 1 dan larik 2 tidak sama. Oleh karena itu, Penjumlahan tidak dapat dilakukan.");
            return;
        }

        for(int a = 0; a < m1.ukuran; a++){
            this.arr[a] = m1.arr[a] + m2.arr[a];
        }
    }
    
    public void output(){
        for(int a : this.arr){
            System.out.print(a + " ");
        }
    }
    
}

public class Soal1{
    static Scanner sc = new Scanner(System.in);
    
    static void menu(){
        int pilihan = 1;
        Logika data = new Logika();
        Logika data2 = new Logika();
        Logika data3 = new Logika();


        while(pilihan != 0){
            System.out.println("\n\n\tMENU");
            System.out.println("====================");
            System.out.println("1. Input Matriks Sebagai Data pilihan 2, 3, dan 4");
            System.out.println("2. Mencari Rata Rata");
            System.out.println("3. Mencari Nilai Maks");
            System.out.println("4. Mencari Nilai Min");
            System.out.println("5. Mencari jlh matriks");
            System.out.println("0. Exit");
            System.out.print("> ");

            pilihan = sc.nextInt();

            switch(pilihan){
                case 1 :
                    data.inputUkuran();
                    data.isiArr();
                    break;
                case 2:
                    System.out.println("Nilai rata-rata = " + data.cariRata2());
                    break;
                case 3:
                    System.out.println("Nilai Maks = " + data.nilaiMax());
                    break;
                case 4:
                    System.out.println("Nilai Min = " + data.nilaiMin());
                    break;
                case 5:          
                    data2.inputUkuran();
                    System.out.println("Matriks 1 : ");
                    data2.isiArr();

                    System.out.println("\nMatriks 2 : ");
                    data3.setUkuran(data2.getUkuran());
                    data3.isiArr();

                    System.out.println("\nHasil Jumlah Matriks : ");
                    data3.jumlahMatriks(data2, data3);
                    data3.output();
                    break;
                case 0:
                    System.out.println("Program selesai!!");                    
                    break;
                default:
                    System.out.println("Pilihan tidak ada diMenu!!");
                    break;
            }
        }
    }
    public static void main(String[] args) {
        menu();
    }
}

File: Soal1.py
 # Nama : Wilson Angelie Tan, Stan F., Theophilus Samuel G.
 # NPM :140810230024, 140810230046, 140810230054
 # Tanggal Buat :13/10/2024
 # Deskripsi :Cari nilai rata-rata nilai, nilai
 #          terbesar dan terkecilmenggunakan array satu dimensi dalam
 #          bentuk OOP 2 class bahasapython
class Logika:        
    def __init__(self, ukuran=10):
        self.__ukuran = ukuran
        self.__arr = [0 for _ in range(self.__ukuran)]
        
    def setUkuran(self, ukuran):
        self.__ukuran = ukuran
        self.__arr = [0 for _ in range(self.__ukuran)]
        
    def inputUkuran(self):
        self.__ukuran = int(input("Masukkan Ukuran Array : "))
        self.__arr = [0 for _ in range(self.__ukuran)]
        
    def getUkuran(self):
        return self.__ukuran
    
    def setArr(self, arr):
        self.__arr = arr
    
    def getArr(self):
        return self.__arr
    
    def isiArr(self):
        for a in range(self.__ukuran):
            self.__arr[a] = int(input(f"Masukkan nilai ke-{a+1} = "))
            
    def cariRata2(self):
        jumlah = 0
        
        for a in self.__arr:
            jumlah += a
        
        return (jumlah/self.__ukuran)
    
    def nilaiMax(self):
        highest = -99999
        
        for a in self.__arr:
            if(highest < a):
                highest = a
                
        return highest
    
    def nilaiMin(self):
        lowest = 99999
        
        for a in self.__arr:
            if(lowest > a):
                lowest = a
                
        return lowest
    
    def jumlahMatriks(self, m1, m2):
        if(m1.__ukuran != m2.__ukuran):
            print("Ukuran Larik 1 dan larik 2 tidak sama. Oleh karena itu, Penjumlahan tidak dapat dilakukan.")
            return
        
        for a in range(m1.__ukuran):
            self.__arr[a] = m1.__arr[a] + m2.__arr[a]
            
    def output(self):
        for a in self.__arr:
            print(a, " ", end="")
            
def menu():
    pilihan = 1
    data = Logika()
    data2 = Logika()
    data3 = Logika()
    
    while(pilihan != 0):
        print("\n\tMENU")
        print("====================")
        print("1. Input Matriks Sebagai Data pilihan 2, 3, dan 4")
        print("2. Mencari Rata Rata")
        print("3. Mencari Nilai Maks")
        print("4. Mencari Nilai Min")
        print("5. Mencari jlh matriks")
        print("0. Exit")
        pilihan = int(input("> "))
        
        if(pilihan == 1):
            data.inputUkuran()
            data.isiArr()
        elif(pilihan == 2):
            print("Nilai rata-rata = ", data.cariRata2())
        elif(pilihan == 3):
            print("Nilai Maks = ", data.nilaiMax())
        elif(pilihan == 4):
            print("Nilai Min = ", data.nilaiMin())
        elif(pilihan == 5):
            data2.inputUkuran()
            print("Matriks 1 : ")
            data2.isiArr()
            
            print("\nMatriks 2 : ")
            data3.setUkuran(data2.getUkuran())
            data3.isiArr()
            
            print("\nHasil Jumlah Matriks : ")
            data3.jumlahMatriks(data2, data3)
            data3.output()
        elif(pilihan == 0):
            print("Program selesai!!")
        else:
            print("Pilihan tidak ada di Menu!!")
            
menu()